src/main.rs

```diff
@@ -128,120 +132,67 @@ fn main() {
-
-                let next = signal
-                    .select2(to_shutdown)
-                    .compat()
-                    .await
-                    .map_err(|_| ())
-                    .expect("Neither stream errors");
-
-                let signal = match next {
-                    future::Either::A((signal, _)) => signal.expect("Signal streams never end"),
-                    // Trigger graceful shutdown if a component crashed, or all sources have ended.
-                    future::Either::B((_to_shutdown, _)) => SIGINT,
-                };
-
-                if signal != SIGHUP {
-                    break signal;
-                }
-
-                // Reload config
-                info!(
-                    message = "Reloading configs.",
-                    path = ?config_paths
-                );
-                let new_config = read_configs(&config_paths);
-
-                trace!("Parsing config");
-                let new_config = handle_config_errors(new_config);
-                if let Some(new_config) = new_config {
-                    match topology
-                        .reload_config_and_respawn(new_config, opts.require_healthy)
-                        .await
-                    {
-                        Ok(true) => (),
-                        Ok(false) => error!("Reload was not successful."),
-                        // Trigger graceful shutdown for what remains of the topology
-                        Err(()) => break SIGINT,
+        let mut signals = signal::signals();
+        let mut sources_finished = topology.sources_finished().compat();
+        let mut graceful_crash = graceful_crash.compat();
+
+        let signal = loop {
+            select! {
+                Some(signal) = signals.next() => {
+                    if signal == SignalTo::Reload{
+                        // Reload config
+                        info!(
+                            message = "Reloading configs.",
+                            path = ?config_paths
+                        );
+                        let new_config = read_configs(&config_paths);
+
+                        trace!("Parsing config");
+                        let new_config = handle_config_errors(new_config);
+                        if let Some(new_config) = new_config {
+                            match topology
+                                .reload_config_and_respawn(new_config, opts.require_healthy)
+                                .await
+                            {
+                                Ok(true) => (),
+                                Ok(false) => error!("Reload was not successful."),
+                                // Trigger graceful shutdown for what remains of the topology
+                                Err(()) => break SignalTo::Shutdown,
+                            }
+                        } else {
+                            error!("Reload aborted.");
+                        }
+                    }else{
```

Comment:
I think this is all I saw. I wonder if the maco confuses rustfmt?

Reply:
It does, outside of the macro it formats it fine, I'll change it in other PR.